app/formats/json/TaskSubmissionFormats.scala

```diff
@@ -116,7 +117,9 @@ object TaskSubmissionFormats {
       (JsPath \ "camera_heading").readNullable[Float] and
       (JsPath \ "camera_pitch").readNullable[Float] and
       (JsPath \ "links").read[Seq[GSVLinkSubmission]] and
-      (JsPath \ "copyright").read[String]
+      (JsPath \ "copyright").read[String] and 
+      (JsPath \ "history").readNullable[Seq[PanoDate]] and
```

Comment:
Ok so that's because in validate & label, our formats of passing the pano histories are slightly different. In validate, we pass all the individual pano history objects in a list of pano history objects, but the thing is that list can just be empty and so we won't actually have any pano histories to add. In label, since individual pano histories are passed inside those panoramas themselves, and sometimes a panorama can come without a history (the case where we already submitted that pano's pano history in the current mission), it's nullable. Let me know your thoughts on restructuring that further!

Reply:
>sometimes a panorama can come without a history (the case where we already submitted that pano's pano history in the current mission), it's nullable

Ohhhh I see. We only include the history once per session. I think that that's okay then. It's more efficient to do it the way you're doing it, but it also feels a bit weird to submit only _almost_ all of the data for the pano... If it's super easy to change such that the history is always sent (especially if it makes the code more simple), then you should probably do that. But I don't consider it to be of the utmost importance, so I am fine with the current implementation :)